src/index.ts

```diff
@@ -88,4 +119,117 @@ internalTask(TASK_COMPILE_SOLIDITY_GET_SOLC_BUILD).setAction(
 
     return wasmCompiler;
   }
-);
\ No newline at end of file
+);
+
+const downloadCompiler = async (downloadPath: string) => {
+  const os = process.platform;
+  const solxLink = await getSolxLink(os);
+  const checksum = await getChecksum(os);
+
+  try {
+    console.log('\x1b[1m\x1b[32m%s\x1b[0m%s', 'Info: ', 'Downloading to: ' + downloadPath);
+    const data = await download(solxLink);
+
+    const hash = crypto.createHash('sha256');
+    const hashResult = hash.update(data).digest('hex');
+
+    if (hashResult !== checksum) {
+      console.error('\x1b[1m\x1b[31m%s\x1b[0m%s', 'Error:', ' Checksum mismatch');
+      process.exit(1);
+    }
+
+    fsExtra.writeFileSync(downloadPath, data);
+    fsExtra.chmodSync(downloadPath, 0o755);
+    console.log('\x1b[1m\x1b[32m%s\x1b[0m%s', 'Info: ', 'Download successful!');
+  } catch (error) {
+    console.error('\x1b[1m\x1b[31m%s\x1b[0m%s', 'Error:', ' Failed to download SOLX');
+    console.error('\x1b[1m\x1b[31m%s\x1b[0m%s', 'Error:', error)
+    process.exit(1);
+  }
+};
+
+const askQuestion = (query: string): Promise<string> => {
+  return new Promise(resolve => rl.question(query, (ans) => {
+    rl.close();
+    resolve(ans);
+  }));
+};
+
+const getLinuxDistribution = async () => {
+  if (ubuntu_version !== undefined) {
+    return ubuntu_version;
+  }
+  console.log("Select your Linux distribution:");
+  console.log("1: Ubuntu 18");
+  console.log("2: Ubuntu 20");
+  console.log("3: Other")
+  console.log("Enter the number of your distribution:");
+
+  const choice = await askQuestion('> ');
+
+  switch (choice.trim().toLowerCase()) {
+    case '1':
+    case 'ubuntu18':
+    case 'ubuntu 18':
+      console.log('\x1b[1m\x1b[31m%s\x1b[0m%s', 'Error:', " Ubuntu 18 download is not supported yet, please compile SolidityX from source");
+      process.exit(1);
+      ubuntu_version = 'ubuntu18'
+      return 'ubuntu18';
+    case '2':
+    case 'ubuntu20':
+    case 'ubuntu 20':
+      ubuntu_version = 'ubuntu20'
+      return 'ubuntu20';
+    case '3':
+    case 'other':
+      console.log('\x1b[1m\x1b[31m%s\x1b[0m%s', 'Error: ', 'Unfortunetly, SolidityX does not support your operating system yet');
+      process.exit();
+    default:
+      console.error('\x1b[1m\x1b[31m%s\x1b[0m%s', 'Error:', ' Invalid selection');
+      process.exit(1);
+  }
+};
+
+const getChecksum = async (os: string) => {
+  switch (os) {
+    case 'darwin':
+      return DARWIN_CHECKSUM;
+    case 'linux':
+      const linuxDistro = await getLinuxDistribution();
+      if (linuxDistro === 'ubuntu18') {
+        return UBUNTU_18_CHECKSUM;
+      } else if (linuxDistro === 'ubuntu20') {
+        return UBUNTU_20_CHECKSUM;
+      }
+      break;
+    case 'win32':
+      console.log('\x1b[1m\x1b[31m%s\x1b[0m%s', 'Error:', " Windows download is not supported yet, please compile SolidityX from source");
```

Comment:
Including a link to the solidity compiler build from source instructions here: https://docs.soliditylang.org/en/latest/installing-solidity.html

Or maybe a link to the SolidityX repo build from source instructions as we continue to build them out for different OS: https://github.com/dominant-strategies/SolidityX?tab=readme-ov-file#the-solidity-contract-oriented-programming-language

Same goes for the rest of the instances where we tell the user `Your OS is not supported yet`

Reply:
Done